spring/src/main/java/org/axonframework/spring/config/NoBeanOfType.java

```diff
@@ -35,6 +35,7 @@
 @Target(ElementType.METHOD)
 @Retention(RetentionPolicy.RUNTIME)
 @Conditional(NoBeanOfType.NoBeanOfTypeDefined.class)
+@Deprecated
```

Comment:
This change warrants an adjustment in the JavaDoc describing why it's deprecated and what (if any) replaces it.